Spigot-Server-Patches/0696-Horse-Api.patch

```diff
@@ -0,0 +1,93 @@
+         this.datawatcher.register(EntityHorseAbstract.bA, Optional.empty());
+     }
+ 
++    public boolean getHorseFlag(int mask) {return t(mask);} // Paper - OBFHELPER
+     protected boolean t(int i) {
+         return ((Byte) this.datawatcher.get(EntityHorseAbstract.bz) & i) != 0;
+     }
+ 
++    public void setHorseFlag(int mask, boolean value) {d(mask, value);} // Paper - OBFHELPER
+     protected void d(int i, boolean flag) {
+         byte b0 = (Byte) this.datawatcher.get(EntityHorseAbstract.bz);
+ 
+@@ -176,10 +178,12 @@ public abstract class EntityHorseAbstract extends EntityAnimal implements IInven
+ 
+     }
+ 
++    public boolean isEatingGrass() {return eZ();} // Paper - OBFHELPER
+     public boolean eZ() {
+         return this.t(16);
+     }
+ 
++    public boolean isRearing() {return fa();} // Paper - OBFHELPER
+     public boolean fa() {
+         return this.t(32);
+     }
+@@ -654,6 +658,7 @@ public abstract class EntityHorseAbstract extends EntityAnimal implements IInven
+ 
+     }
+ 
++    public void setEatingGrass(boolean eating) {x(eating);} // Paper - OBFHELPER
+     public void x(boolean flag) {
+         this.d(16, flag);
+     }
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftAbstractHorse.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftAbstractHorse.java
+index 5b66165471197aad57e23f9a6669b11f25a5e4f2..9879db6e831591228f8016e3c6ebf0563ee0b98d 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftAbstractHorse.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftAbstractHorse.java
+@@ -5,7 +5,6 @@ import net.minecraft.world.entity.ai.attributes.GenericAttributes;
+ import net.minecraft.world.entity.animal.horse.EntityHorseAbstract;
+ import org.apache.commons.lang.Validate;
+ import org.bukkit.craftbukkit.CraftServer;
+-import org.bukkit.craftbukkit.inventory.CraftInventoryAbstractHorse;
+ import org.bukkit.craftbukkit.inventory.CraftSaddledInventory;
+ import org.bukkit.entity.AbstractHorse;
+ import org.bukkit.entity.AnimalTamer;
+@@ -105,4 +104,36 @@ public abstract class CraftAbstractHorse extends CraftAnimals implements Abstrac
+     public AbstractHorseInventory getInventory() {
+         return new CraftSaddledInventory(getHandle().inventoryChest);
+     }
++
++    // Paper start - Horse API
++    @Override
++    public boolean isEatingGrass() {
++        return getHandle().isEatingGrass();
++    }
++
++    @Override
++    public void setEatingGrass(boolean eating) {
++        getHandle().setEatingGrass(eating);
++    }
++
++    @Override
++    public boolean isRearing() {
++        return getHandle().isRearing();
++    }
++
++    @Override
++    public void setRearing(boolean rearing) {
++        getHandle().setHorseFlag(32, rearing);
++    }
++
++    @Override
++    public boolean isEating() {
++        return getHandle().getHorseFlag(64);
++    }
++
++    @Override
++    public void setEating(boolean eating) {
++        getHandle().setHorseFlag(64, eating);
```

Comment:
Modify the lines that specify these integers to use static fields with them instead so we can reuse here.